source/websocket.c

```diff
@@ -216,12 +233,26 @@ void s_unlock_synced_data(struct aws_websocket *websocket) {
     (void)err;
 }
 
-struct aws_channel_handler *aws_websocket_handler_new(const struct aws_websocket_handler_options *options) {
+struct aws_websocket *aws_websocket_handler_new(const struct aws_websocket_handler_options *options) {
```

Comment:
talked offline, I think the constructor shouldn't be doing the channel operations like this, and we should at least break the orchestration up between functions. Your call though.

Reply:
Agreed that this does not separate concerns, but given how the channel API works, it was the most RAII way of doing things I could come up with, where the thing returned from the function is ready to go, and if there's an error then things end up in the same state they started.

The Channel API could really use a revision pass that makes a clean setup possible.